packages/data-context/src/sources/MigrationDataSource.ts

```diff
@@ -305,7 +321,13 @@ export class MigrationDataSource {
     return this._step
   }
 
-  setStep (step: MIGRATION_STEP) {
+  async setStep (step: MIGRATION_STEP) {
+    if (this.componentTestingMigrationWatcher) {
```

Comment:
What happens if I leave the step wthout setting the step? Say I click the projects link to go back to global mode. Would the watcher still close? Should we call a mutation on unmount to make sure that it gets cleaned up?